### PR TITLE
OP-1016 Upgrade to java 17 and upgraded maven

### PR DESCRIPTION
**Pull request by Th0rst3nH**
[OP-1019](https://openhospital.atlassian.net/browse/OP-1019) Java 17 upgrade for OH-api

[OP-1019]: https://openhospital.atlassian.net/browse/OP-1019?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -90,44 +90,25 @@
...
             <groupId>io.jsonwebtoken</groupId>
             <artifactId>jjwt-impl</artifactId>
             <version>0.11.5</version>
-            <scope>runtime</scope>
-        </dependency>
-        <dependency>
-            <groupId>io.jsonwebtoken</groupId>
-            <artifactId>jjwt-jackson</artifactId>
-            <version>0.11.5</version>
-            <scope>runtime</scope>
```
Comment (mwithi) [inline pom.xml:117]: Looks like both jjwt-api and jjwt-jackson are needed and used in `src/main/java/org/isf/security/jwt/TokenProvider.java`

File: openapi/oh.yaml
Comment (mwithi) [inline openapi/oh.yaml:1]: If there are no breaking changes in the apidoc we can revert these changes

**Review by dbmalkovsky — APPROVED** [review]
LGTM; ran `mvn clean test` without error.
